jsaddle/src/Language/Javascript/JSaddle/Types.hs

```diff
@@ -786,18 +798,35 @@ instance MonadError JSVal JSM where
           sendReq Req_FinishTry
         action = (runReaderT (unJSM (a <* end)) env)
           `catch` (\e@(SomeException _) -> do
+                      killChild
                       case fromException e of
                         Just (_ :: AsyncCancelled) -> pure ()
                         Nothing -> putMVar asyncExceptionVar e
                       throwIO e)
+        killChild = do
+          mChildTryId <- readTVarIO newChildTryIdTVar
+          forM_ mChildTryId $ \childTryId -> do
+            mChildTryMVar <- atomically $ do
+              currentTries <- readTVar tries
+              writeTVar tries $! M.delete childTryId currentTries
+              return $ M.lookup childTryId currentTries
+            forM_ mChildTryMVar $ \v ->
+              putMVar v $ Left $ primToJSVal $ PrimVal_String "Parent Try received an exception."
+
     tryResult <- JSM $ liftIO $ withAsync action $ \aa -> do
       void $ tryPutMVar sendReqsBatchVar ()
+      -- Set the childTryIdVar for the parent, so that if the parent dies, this try
+      -- recieves the finishVar Left
+      liftIO $ atomically $ modifyTVar' childTryIdTVar $ \case
+        Just _ -> error "childTryIdTVar already has a try in it"
```

Comment:
Does this error need to be fatal? It may not be a problem if this case is impossible, but I was wondering if two simultaneous catchError's could land here.

Reply:
I suppose the alternative is potentially throwing away one of the errors which is not much better though.

Reply:
Got rid of this